packages/fast-components-react-base/src/auto-suggest/auto-suggest.spec.tsx

```diff
@@ -0,0 +1,286 @@
+import * as React from "react";
+import * as Adapter from "enzyme-adapter-react-16";
+import { configure, mount, render, shallow } from "enzyme";
+import AutoSuggest, { AutoSuggestUnhandledProps } from "./auto-suggest";
+import ListboxItem from "../listbox-item";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+
+/*
+ * Configure Enzyme
+ */
+configure({ adapter: new Adapter() });
+
+const itemA: JSX.Element = <ListboxItem id="a" value="a" displayString="a" />;
+const itemB: JSX.Element = <ListboxItem id="b" value="b" displayString="ab" />;
+const itemC: JSX.Element = <ListboxItem id="c" value="c" displayString="abc" />;
+
+describe("auto select", (): void => {
+    test("should have a displayName that matches the component name", () => {
+        expect((AutoSuggest as any).name).toBe(AutoSuggest.displayName);
+    });
+
+    test("should not throw if managedClasses are not provided", () => {
+        expect(() => {
+            shallow(<AutoSuggest listboxId="listboxId" />);
+        }).not.toThrow();
+    });
+
+    test("should implement unhandledProps", (): void => {
+        const unhandledProps: AutoSuggestUnhandledProps = {
+            "aria-label": "label",
+        };
+
+        const rendered: any = shallow(
+            <AutoSuggest listboxId="listboxId" {...unhandledProps} />
+        );
+
+        expect(rendered.first().prop("aria-label")).toEqual("label");
+    });
+
+    test("initial default state is correct", (): void => {
+        const rendered: any = mount(
+            <AutoSuggest listboxId="listboxId">
+                {itemA}
+                {itemB}
+                {itemC}
+            </AutoSuggest>
+        );
+
+        expect(rendered.state("value")).toBe("");
+        expect(rendered.state("focusedItem")).toBe(null);
+        expect(rendered.state("isMenuOpen")).toBe(false);
+    });
+
+    test("default input region attributes are set correctly", (): void => {
+        const rendered: any = mount(
+            <AutoSuggest listboxId="listboxId" label="test-label" disabled={true}>
+                {itemA}
+                {itemB}
+                {itemC}
+            </AutoSuggest>
+        );
+
+        const input: any = rendered.find("input");
+        expect(input.prop("aria-label")).toEqual("test-label");
+        expect(input.prop("aria-autocomplete")).toEqual("both");
+        expect(input.prop("aria-activedescendant")).toEqual(null);
+        expect(input.prop("aria-owns")).toEqual(null);
+        expect(input.prop("aria-controls")).toEqual(null);
+        expect(input.prop("disabled")).toBe(true);
+        expect(input.prop("role")).toEqual("combobox");
+    });
+
+    test("input region attributes change correctly as options list navigated", (): void => {
```

Comment:
I think we want more detail here, perhaps in this case since attributes are changing during list navigation we should consider breaking this up based on behavior

Reply:
made tests more "atomic"